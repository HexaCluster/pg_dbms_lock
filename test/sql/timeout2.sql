-- Create an exlusive lock in the first session
DO $$
DECLARE
    v_result     integer;
    dbms_lock_x_mode integer := 6;
BEGIN

    -- should fail with timeout
    v_result := dbms_lock.request(123, dbms_lock_x_mode, 1);

    IF v_result <> 0 THEN
        RAISE NOTICE '%', (
           case 
              when v_result=1 then 'Timeout'
              when v_result=2 then 'Deadlock'
              when v_result=3 then 'Parameter Error'
              when v_result=4 then 'Already owned'
              when v_result=5 then 'Illegal Lock Handle'
            end);
    END IF;

END;
$$;

SELECT objid, mode FROM pg_locks WHERE objid IS NOT NULL AND locktype = 'advisory' AND database = (SELECT oid FROM pg_database WHERE datname = current_database()) ORDER BY objid;

