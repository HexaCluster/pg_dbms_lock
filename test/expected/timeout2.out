psql:test/sql/timeout2.sql:23: WARNING:  Could not acquire advisory lock in 1 seconds
psql:test/sql/timeout2.sql:23: NOTICE:  Timeout
DO
 objid |     mode      
-------+---------------
   123 | ExclusiveLock
(1 row)

